enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,751 @@
+choices.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The changes in this enhancement are focused on OpenShift standalone
+topologies. Hypershift, and the hosted control plane topologies based
+on it, use Cluster API in a differnt configuration than standalone. As
+such, these clusters have their Cluster Autoscaler configured in a
+manner that does not depend on the Cluster Autoscaler Operator.
+
+Please see the Hypershift enhancement on [Node lifecycle][hcp-nl] for more
+information about the Cluster Autoscaler and its relationship to hosted
+control plane topologies.
+
+#### Standalone Clusters
+
+The changes in this enhancement are specifically meant for standalone clusters.
+These changes are not meant for deployment to Hypershift and hosted control plane
+topologies as they do not use the Cluster Autosaler Operator, and have a different
+configuration for the Cluster Autoscaler and Cluster API.
+
+#### Single-node Deployments or MicroShift
+
+The changes in this enhancement are not meant for single-node deployments. The
+Cluster Autoscaler and Cluster Autoscaler Operator are not deployed in single-node
+clusters, and as such this feature does not have relevance on that topology.
+
+### Implementation Details/Notes/Constraints
+
+The implementation changes for this enhancement will be done in the Cluster
+Autoscaler, Cluster Autoscaler Operator, and Cluster API MachineSet Sync Controller
+projects. Most of the details are internal changes to how Kubernetes resource data
+is applied and updated. The user-facing changes will be confined to the MachineAutoscaler
+resource associated with the Cluster Autoscaler Operator.
+
+#### Cluster Autoscaler Changes
+
+To ensure proper operation and reduce the complexity of integration, the Cluster
+Autoscaler will be configured to interface with Cluster API MachineSet resources.
+This change is primarily a change to the running configuration of the Cluster Autoscaler.
+
+Due to historical work on the Cluster Autoscaler and the sibling relationship of the
+Machine API and Cluster API, the Cluster Autoscaler customization has been reduced to
+environment variables and a few carried patches to express the Machine API functionality.
+Reconfiguring the Cluster Autoscaler for Cluster API will mean that some of the carried
+patches can be removed, and a change in the deployment configuration.
+
+#### MachineAutoscaler Spec Changes
+
+The main implementation detail of this enhancement that users will interact with is
+how the Cluster Autoscaler Operator will locate the MachineSet resources referenced by
+the MachineAutoscaler resource.
+
+Previously, only a Machine API MachineSet (i.e. a `MachineSet` kind in the
+`machine.openshift.io` API group) would be valid target of the `.spec.scaleTargetRef`
+field. After this enhancement is implemented, users may specify either a Machine
+API MachineSet or a Cluster API MachineSet in the `.spec.scaleTargetRef` field.
```

Comment:
Given every MAPI machineset will have a CAPI mirror, when CAPI mirroring is enabled, do we actually need to care about the MAPI side?

Reply:
i think we care to the extent that we want to allow users to continue using MAPI resources as the `scaleTargetRef`. basically, not breaking existing MachineAutoscalers.